evap/staff/templates/staff_semester_view.html

```diff
@@ -549,23 +553,15 @@ <h3>
     {% trans 'Activate reward points' as action_text %}
     {% include 'confirmation_modal.html' with modal_id='activateRewardPointsModal' title=title question=question action_text=action_text btn_type='primary' %}
     <script type="text/javascript">
-        function activateRewardPointsModalAction(dataId) {
-            $.ajax({
-                type: "POST",
-                url: "{% url 'rewards:semester_activation' semester.id 'on' %}",
-                data: {},
-                success: function(){ location.reload(); },
-                error: function(){ window.alert("{% trans 'The server is not responding.' %}"); }
-            });
+        function setRewardPointActivationStatus(status) {
+            document.getElementById("activation_status").value = status;
+            document.getElementById("form_activation_status").submit();
```

Comment:
This is a bit odd, persisting the state in the DOM like this. I think the optimal modeling would link the two submit buttons to the form and give them `name` and `value` (if we have `<button type="submit" name="foo" value="bar">`, clicking the button will submit with the additional data `foo=bar`; using `form=theform` on the button makes it possible for it to be outside of the `form` tag). This works directly with the deactivation-button, however for the activation, the real button is inside the modal and we currently don't have a way to modify this button from the outside (it would also be prone to error). I am not sure what the best solution is here, currently I am think that we should have two buttons with `form`, `name` and `value` where we currently have the button that triggers `deactivateRewardPoints`, make the activation one hidden and let the modal-action click this hidden button. But that is also a bit weird. @He3lixxx Do you a better idea?

Reply:
Not really.

In general, I think our modals should support adding additional POST data to the submission, we may just want to add that.